packages/tools/examples/contourApi/index.ts

```diff
@@ -0,0 +1,530 @@
+          holeResult.holeIndexes.forEach((holeIndex) => {
+            const polyline = allPolylines[holeIndex];
+            const annotation = annotationPolylineMap.get(polyline);
+            if (annotation) {
+              // Set annotation style to red
+              const styles = {
+                color: 'rgb(255, 0, 0)', // Red color
+              };
+
+              csToolsAnnotation.config.style.setAnnotationStyles(
+                annotation.annotationUID,
+                styles
+              );
+            }
+          });
+          // select the polyline with holes
+          const polyline = allPolylines[holeResult.contourIndex];
+          const annotation = annotationPolylineMap.get(polyline);
+          if (annotation) {
+            csToolsAnnotation.selection.setAnnotationSelected(
+              annotation.annotationUID
+            );
+          }
+        });
+
+        // Re-render to show the color changes
+        const renderingEngine = getRenderingEngine(renderingEngineId);
+        renderingEngine.render();
+      }
+    } catch (error) {
+      console.error('Error finding contour holes:', error);
+    }
+  },
+});
+
+addButtonToToolbar({
+  title: 'Supersample Polylines',
+  onClick: () => {
+    const annotations = cornerstoneTools.annotation.state.getAllAnnotations();
+
+    if (annotations.length === 0) {
+      console.log('No annotations found. Please draw some contours first.');
+      return;
+    }
+
+    const renderingEngine = getRenderingEngine(renderingEngineId);
+
+    annotations.forEach((annotation, index) => {
+      if (annotation?.data?.contour?.polyline) {
+        const viewport = getViewportDisplayingAnnotation(annotation);
+        if (!viewport) {
+          return;
+        }
+        const originalPolyline = convertContourPolylineToCanvasSpace(
+          annotation.data.contour.polyline,
+          viewport
+        );
+
+        console.log(`Annotation ${index + 1}:`);
+        console.log('Original polyline points:', originalPolyline.length);
+
+        try {
+          // Supersample the polyline with target spacing of 1.0
+          const supersampledPolyline = supersamplePolyline2D(
+            originalPolyline,
+            0.1
+          );
+          console.log(
+            'Supersampled polyline points:',
+            supersampledPolyline.length
+          );
+
+          // Update the annotation with the supersampled polyline
+          annotation.data.contour.polyline = convertContourPolylineToWorld(
+            supersampledPolyline,
+            viewport
+          );
+        } catch (error) {
+          console.error('Error supersampling polyline:', error);
+        }
+      }
+    });
+
+    // Re-render to show the updated polylines
+    renderingEngine.renderViewports(viewportIds);
+  },
+});
+
+addButtonToToolbar({
+  title: 'Remove Small Islands',
+  onClick: () => {
+    const annotations = cornerstoneTools.annotation.state.getAllAnnotations();
```

Comment:
cornerstoneTools.segmentation.utilities.removeContourIslands({ segmentationId, segmentIndex  options: { epsilon })

Reply:
it should remove them automatically the following logic should not exist in the example